docs/security_self_assessment.md

```diff
@@ -0,0 +1,1588 @@
+
+## Testing Controls Methodology
+
+RKE2 installs Kubernetes services via static pods. Configuration is defined by arguments passed to the pod at the time of initialization.
+
+Where control audits differ from the original CIS benchmark, the audit commands specific to RKE2 are provided for testing. When performing the tests, you will need access to the command line on the hosts of the RKE2 nodes. The commands also make use of the the `jq` and `kubectl` (with valid config) tools to and are required in the testing and evaluation of test results.
+
+## Controls
+
+### 1 Master Node Security Configuration
+
+#### 1.1 Master Node Configuration Files
+
+*1.1.1* Ensure that the API server pod specification file permissions are set to `644` or more restrictive (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for the API server. All configuration is passed in as arguments at container run time.
+
+*1.1.2* Ensure that the API server pod specification file ownership is set to `root:root` (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for the API server. All configuration is passed in as arguments at container run time.
+
+*1.1.3* Ensure that the controller manager pod specification file permissions are set to `644` or more restrictive (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for the controller manager. All configuration is passed in as arguments at container run time.
+
+*1.1.4* Ensure that the controller manager pod specification file ownership is set to `root:root` (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for the controller manager. All configuration is passed in as arguments at container run time.
+
+*1.1.5* Ensure that the scheduler pod specification file permissions are set to `644` or more restrictive (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for the scheduler. All configuration is passed in as arguments at container run time.
+
+*1.1.6* Ensure that the scheduler pod specification file ownership is set to `root:root` (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for the scheduler. All configuration is passed in as arguments at container run time.
+
+*1.1.7* Ensure that the etcd pod specification file permissions are set to `644` or more restrictive (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for etcd. All configuration is passed in as arguments at container run time.
+
+*1.1.8* Ensure that the etcd pod specification file ownership is set to `root:root` (Scored)
```

Comment:
I dont really know why we skipped 1.1.9 and 1.1.10 in the rancher guide, but I dont think we should skip it here.

Reply:
They are the tests for the CNI file perms and ownership.
I guess they are skipped because the are "manual" and it is hard to apply the control to RKE1.
But, we should still include the controls here with a remediation that explains that rke2 deploys canal as the cni by default using helm and thus there is is not explicit config file.

Reply:
I followed up with Bill on his position around what controls we show/don't show

Going forward we are going to show all controls. If one doesn't apply to us, we have to explain in the remidation why.